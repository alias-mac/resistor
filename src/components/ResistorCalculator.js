/*
 * Copyright (c) 2016 Filipe Guerra
 * https://github.com/alias-mac/resistor
 *
 * For the full copyright and license information, please view the LICENSE
 * file that was distributed with this source code.
 */

import React from 'react';
import PropTypes from 'prop-types';

import '../styles/resistor-calc.scss';

import Resistor from '../Resistor';
import ResistorForm from './ResistorForm';
import ResistorValue from './ResistorValue';
import ResistorSvg from './ResistorSvg';

class ResistorCalculator extends React.Component {

  constructor(props) {
    super(props);

    this.state = {
      resistor: new Resistor({
        bands: 4,
        digit1: 1,
        digit2: 0,
        multiplier: 0,
      }),
    };

    this.onChange = this.onChange.bind(this);
  }

  onChange(prop, value) {
    const resistor = this.state.resistor;
    resistor.set(prop, value);
    this.setState({ resistor });
  }

  render() {
    return (
      <div className="resistor-calc">
        <h2>Resistor Calculator</h2>
        <h4>Calculates values from color codes</h4>
        <this.props.form model={this.state.resistor} onChange={this.onChange} />
        <ResistorValue
          className="resistor-value" formatOptions={{ short: true }}
          model={this.state.resistor}
        />
        <ResistorSvg model={this.state.resistor} />
      </div>
    );
  }
}

ResistorCalculator.propTypes = {
  form: PropTypes.func,
};

ResistorCalculator.defaultProps = {
  form: ResistorForm,
};

export default ResistorCalculator;
