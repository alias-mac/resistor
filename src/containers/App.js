/*
 * Copyright (c) 2016 Filipe Guerra
 * https://github.com/alias-mac/resistor
 *
 * For the full copyright and license information, please view the LICENSE
 * file that was distributed with this source code.
 */

import React from 'react';
import Resistor from '../Resistor';
import ResistorCalculator from '../components/ResistorCalculator';

class App extends React.Component {

  constructor(props) {
    super(props);

    this.state = {
      resistor: new Resistor({
        bands: 4,
        digit1: 1,
        digit2: 0,
        multiplier: 0,
      }),
    };

    this.onChange = this.onChange.bind(this);
  }

  onChange(prop, value) {
    const resistor = this.state.resistor;
    resistor.set(prop, value);
    this.setState({ resistor });
  }

  render() {
    return (
      <div>
        <ResistorCalculator model={this.state.resistor} onChange={this.onChange} />
      </div>
    );
  }
}

export default App;
