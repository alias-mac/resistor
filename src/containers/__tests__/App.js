/*
 * Copyright (c) 2016 Filipe Guerra
 * https://github.com/alias-mac/resistor
 *
 * For the full copyright and license information, please view the LICENSE
 * file that was distributed with this source code.
 */

/* eslint-disable react/prop-types */

import React from 'react';
import { shallow } from 'enzyme';

import ResistorCalculator from 'components/ResistorCalculator';
import App from '../App';

describe('App', function () {

  it('should render with default state', function () {

    const component = shallow(<App />);

    expect(component).toMatchSnapshot();
  });

  it('should update resistor state', function () {

    const component = shallow(<App />);
    const resistorCalc = component.find(ResistorCalculator);

    // initial state (4 bands)
    expect(component).toMatchSnapshot();


    resistorCalc.props().onChange('bands', '3');
    // updated to 3 bands
    expect(component).toMatchSnapshot();

    resistorCalc.props().onChange('bands', '5');
    resistorCalc.props().onChange('digit1', '3');
    resistorCalc.props().onChange('digit2', '5');
    resistorCalc.props().onChange('multiplier', '2');
    // updated to 5 bands with different values and multiplier
    expect(component).toMatchSnapshot();

    resistorCalc.props().onChange('bands', '6');
    resistorCalc.props().onChange('digit3', '7');
    resistorCalc.props().onChange('tolerance', '0.5');
    resistorCalc.props().onChange('tempCoef', '100');
    // updated to 6 bands
    expect(component).toMatchSnapshot();
  });
});
