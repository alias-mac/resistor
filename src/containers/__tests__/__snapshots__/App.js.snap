// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App should render with default state 1`] = `
<div>
  <ResistorCalculator
    model={
      Resistor {
        "attributes": Object {
          "bands": 4,
          "digit1": 1,
          "digit2": 0,
          "digit3": 0,
          "multiplier": 0,
          "tempCoef": 250,
          "tolerance": 10,
        },
      }
    }
    onChange={[Function]}
  />
</div>
`;

exports[`App should update resistor state 1`] = `
<div>
  <ResistorCalculator
    model={
      Resistor {
        "attributes": Object {
          "bands": 4,
          "digit1": 1,
          "digit2": 0,
          "digit3": 0,
          "multiplier": 0,
          "tempCoef": 250,
          "tolerance": 10,
        },
      }
    }
    onChange={[Function]}
  />
</div>
`;

exports[`App should update resistor state 2`] = `
<div>
  <ResistorCalculator
    model={
      Resistor {
        "attributes": Object {
          "bands": 3,
          "digit1": 1,
          "digit2": 0,
          "digit3": 0,
          "multiplier": 0,
          "tempCoef": 250,
          "tolerance": 10,
        },
      }
    }
    onChange={[Function]}
  />
</div>
`;

exports[`App should update resistor state 3`] = `
<div>
  <ResistorCalculator
    model={
      Resistor {
        "attributes": Object {
          "bands": 5,
          "digit1": 3,
          "digit2": 5,
          "digit3": 0,
          "multiplier": 2,
          "tempCoef": 250,
          "tolerance": 10,
        },
      }
    }
    onChange={[Function]}
  />
</div>
`;

exports[`App should update resistor state 4`] = `
<div>
  <ResistorCalculator
    model={
      Resistor {
        "attributes": Object {
          "bands": 6,
          "digit1": 3,
          "digit2": 5,
          "digit3": 7,
          "multiplier": 2,
          "tempCoef": 100,
          "tolerance": 0.5,
        },
      }
    }
    onChange={[Function]}
  />
</div>
`;
